src/masternodes/anchors.cpp

```diff
@@ -217,17 +267,54 @@ CAnchor CAnchorAuthIndex::CreateBestAnchor(CTxDestination const & rewardDest) co
             curHeight = it->height;
             curSignHash = it->GetSignHash();
 
+            // Post fork anchor auth only have one team member. Check it is present here before reading from it!
+            if (ChainActive().Height() >= Params().GetConsensus().DakotaHeight && it->nextTeam.size() == 1)
+            {
+                // Team data reference
+                const CKeyID& teamData = *it->nextTeam.begin();
+
+                uint64_t anchorCreationHeight;
+                std::shared_ptr<std::vector<unsigned char>> prefix;
+
+                // Check this is post-fork anchor auth
+                if (GetAnchorEmbeddedData(teamData, anchorCreationHeight, prefix))
+                {
+                    // Get anchor team at time of creating this auth
+                    auto team = pcustomcsview->GetAuthTeam(anchorCreationHeight);
+                    if (team) {
+                        // Now we can set the appropriate quorum size
+                        quorum = GetMinAnchorQuorum(*team);
+                    } else {
+                        // Should not get here but set max size if we do
+                        quorum = 1 + (Params().GetConsensus().mn.anchoringTeamSize * 2) / 3;
+                    }
+                }
+            }
+
             // we doesn't choose here between "equal" valid auths (by max quorum nor by prevTeam or anything else)
             auto count = list.count(std::make_tuple(curHeight, curSignHash));
+
             if (count >= quorum) {
-                KList::iterator it0, it1;
+                KList::iterator it0, it0Copy, it1;
                 std::tie(it0,it1) = list.equal_range(std::make_tuple(curHeight, curSignHash));
-                for (uint32_t i = 0; i < quorum && it0 != it1; ++i, ++it0) {
-                    LogPrintf("auths: pick up %d, %s, %s\n", it0->height, it0->blockHash.ToString(), it0->GetHash().ToString());
-                    /// @todo do we need for an extra check of the auth signature here?
-                    freshestConsensus.push_back(*it0);
+
+                uint32_t validCount{0};
+                it0Copy = it0;
+                for (uint32_t i{0}; i < quorum && it0Copy != it1; ++i, ++it0Copy) {
+                    // ValidateAuth called here performs extra checks with SPV enabled.
+                    if (ValidateAuth(*it0)) {
```

Comment:
It should be `it0Copy` ?

Reply:
Good spot. It was originally it0 in that loop, failed to update the arg!